src/messages/LabeledMsg.ts

```diff
@@ -57,7 +66,12 @@ export interface PartialMsg<T, V> {
 export function applyProtocol<T, V>(
   partialMsg: PartialMsg<T, V>
 ): LabeledMsg<T, V> {
-  return { ...partialMsg, protocol: API_PROTOCOL, version };
+  return {
+    direction: undefined,
```

Comment:
My initial inclication here would be to require a direction when calling `applyProtocol` rather than leaving it undefined. Or perhaps to create separate `applyHostProtocol` and `applyClientProtocol` messages

Reply:
I tried that at first but then realized that applyProtocol is only used by the client. Oh, I can rename it to applyClientProtocol and then have the host do its own thing.

Reply:
Oh, maybe it should be done on host too, since host is currently setting version to 'unknown' through the decoders

Reply:
I changed it to use the protocols instead of manually setting direction, and it sets version now too.